lib/cretonne/src/verifier/mod.rs

```diff
@@ -409,6 +411,33 @@ impl<'a> Verifier<'a> {
                             got);
             }
         }
+        // We also verify if the postorder defined by `DominatorTree` is sane
+        for (index, (&true_ebb, &test_ebb)) in
+            self.domtree
+                .cfg_postorder()
+                .iter()
+                .zip(domtree.cfg_postorder().iter())
+                .enumerate() {
+            if true_ebb != test_ebb {
+                return err!(test_ebb,
+                            "invalid domtree, postorder ebb number {} should be {}, got {}",
+                            index,
+                            true_ebb,
+                            test_ebb);
+            }
+        }
+        // We verify rpo_cmp on pairs of adjacent ebbs in the postorder
+        for (&prev_ebb, &next_ebb) in self.domtree.cfg_postorder().iter().adjacent_pairs() {
+            match domtree.rpo_cmp(prev_ebb, next_ebb, &self.func.layout) {
+                Ordering::Greater => (),
```

Comment:
I don't think a match is necessary here: `if rpo_cmp(...) != Ordering::Greater`.